docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,156 @@
+
+## Privacy
+
+A memo could include sensitive or personal information about either the sender, the receiver or the transaction itself. In order to maintain privacy, each file containing a memo must be locally encrypted before uploaded to the external service. In the case of an unauthorized access to the external service, memo data will be unreadable. The user must acknowledge, however, that connecting to an external service will be at his/her own risk. If the user loses access to his/her account in the external service, the files could expose metadata about transactions history based on file creation dates.
+
+## Encryption
+
+To safely store data in the external service we have to use encryption and digital signatures. However, before going any further we must notice an important limitation about **hardware wallets**. For security reasons, hardware wallets only provide access to the device's public keys, **NOT** private keys. Then, the proposed approach relies on using solely the *account* public key (available in all wallets) as input to create a new private key chain. More specifically:
+
+* Take the **Public Deriver** public key (*aka* account public key) and hash it with `Blake2b`.
+* Use the result from the previous step as a seed to a key derivation function, which will generate a new key pair. 
+* Use a public key child derivation function with the key pair from the previous step to derive 2 new child keys.
+* From the resulting key pairs, use one to encrypt and the other to sign.
+
+For both encrypting and signing we refer the reader to the corresponding Yoroi specifications for [encryption](https://github.com/Emurgo/yoroi-frontend/blob/737595fec5a89409aacef827d356c9a1605515c0/docs/specs/code/ENCRYPT.md) and [message signing](https://github.com/Emurgo/yoroi-frontend/blob/336f763a7b7085e2887b4965a979ccd24719787a/docs/specs/code/SIGNING.md). More specifically, the signing specification describes an adapted version of the [CBOR](https://tools.ietf.org/html/rfc7049) Object Signing and Encryption ([COSE](https://tools.ietf.org/html/rfc8152)) specification. In our particular case, we must consider the following assumptions:
+
+* A1: We sign with a single key (SIGN_TAG = 18)
+* A2: We use the `sign` and `verify` methods provided by Cardano WASM bindings.
+* A3: Based in A2 we don't need to include the optional parameteres described in point `P1` (from the signing spec) as we already know which algorithm is used.
+* A4: Encryption with the recipient's public key (COSE_Encrypt) is better suited for multiple receivers. For the sake of simplicty, we will use password-based encryption (COSE_Encrypt0) where the ciphertext is obtained as described in the Yoroi [encryption spec](https://github.com/Emurgo/yoroi-frontend/blob/737595fec5a89409aacef827d356c9a1605515c0/docs/specs/code/ENCRYPT.md) with the hash of the shared secret as the password.
```

Comment:
> with the hash of the shared secret as the password

I think this is wrong since the shared secret is meant to deal with encryption using "Public key based encryption". We can generate a pubkey for every memo and use technique if that's what you want, but you mention using the "password-based encryption" instead. Which do you want?

Reply:
Maybe I didn't explain it well. I took the idea from the [signing spec](https://github.com/Emurgo/yoroi-frontend/pull/892#discussion_r312735229). *"single-receiver pubkey-based encryption with COSE_Encrypt0"* 🤔  I didn't elabore it further thinking this was eventually going to be updated in the signing spec.